man/hyprdim.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH hyprdim 1  "hyprdim 2.2.1" 
.SH NAME
hyprdim \- Automatically dim windows in Hyprland when switching between them.
.SH SYNOPSIS
\fBhyprdim\fR [\fB\-s\fR|\fB\-\-strength\fR] [\fB\-d\fR|\fB\-\-duration\fR] [\fB\-f\fR|\fB\-\-fade\fR] [\fB\-b\fR|\fB\-\-bezier\fR] [\fB\-p\fR|\fB\-\-persist\fR] [\fB\-n\fR|\fB\-\-no\-dim\-when\-only\fR] [\fB\-i\fR|\fB\-\-ignore\-entering\-special\fR] [\fB\-I\fR|\fB\-\-ignore\-leaving\-special\fR] [\fB\-D\fR|\fB\-\-dialog\-dim\fR] [\fB\-v\fR|\fB\-\-verbose\fR] [\fB\-h\fR|\fB\-\-help\fR] [\fB\-V\fR|\fB\-\-version\fR] 
.SH DESCRIPTION
.PP
hyprdim is a daemon that automatically dims windows in Hyprland[1] when
switching between them. It works by setting the dim_inactive[2] variable
and changing dim_strength[2] based on whether windows should be dimmed
or not. This enables hyprdim to have smooth dim animations. Additionally,
hyprdim makes use of threads to keep track of new window events. This
enables hyprdim to only disable dim once a user has been inactive long
enough.
.PP
hyprdim is written in Safe Rust[3] and is available under the GPL license[4].
Anyone is free to study the software and expand upon it. The source code is
available here[5].
.PP
[1]: https://hyprland.org/
.PP
[2]: https://wiki.hyprland.org/Configuring/Variables/
.PP
[3]: https://doc.rust\-lang.org/nomicon/meet\-safe\-and\-unsafe.html
.PP
[4]: https://raw.githubusercontent.com/donovanglover/hyprdim/master/LICENSE
.PP
[5]: https://github.com/donovanglover/hyprdim
.PP
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.
.PP
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
.SH OPTIONS
.TP
\fB\-s\fR, \fB\-\-strength\fR=\fISTRENGTH\fR [default: 0.4]
A value from 0 (no dim) to 1 (maximum dim)

Note that negative numbers such as \-1 and \-5 are also supported for "light dim".
.TP
\fB\-d\fR, \fB\-\-duration\fR=\fIDURATION\fR [default: 800]
How many milliseconds to wait before removing dim
.TP
\fB\-f\fR, \fB\-\-fade\fR=\fIFADE\fR [default: 7]
Fade animation speed from 0 (instantaneous) to 255 (very slow)

The slower the fade animation speed is, the higher the duration should be before removing dim.
.TP
\fB\-b\fR, \fB\-\-bezier\fR=\fIBEZIER\fR [default: default]
Bezier curve used for the animation
.TP
\fB\-p\fR, \fB\-\-persist\fR
Prevent dim_inactive from being disabled by `hyprctl reload` etc

Alternatively, set dim_inactive to true in hyprland.conf.
.TP
\fB\-n\fR, \fB\-\-no\-dim\-when\-only\fR
Don\*(Aqt dim when switching to a workspace that only has one visible window

Usually when you use dim in Hyprland, windows are dimmed even when switching between workspaces. This can be annoying if you constantly switch between two different workspaces that each consist of a single window.

Note that this option also takes into account whether a window is fullscreen or not since a workspace with multiple windows and an active one fullscreen makes dimming redundant.

Note that this option also doesn\*(Aqt dim when switching between a special workspace that has only one window. In the future if Hyprland supports fullscreen windows inside special workspaces the codebase will need to be updated as appropriate. Please file an issue or make a pull request if this occurs and there hasn\*(Aqt been an update yet.
.TP
\fB\-i\fR, \fB\-\-ignore\-entering\-special\fR
Don\*(Aqt dim when opening a special workspace

This is useful when you like toggling between a special workspace that has a transparent window and don\*(Aqt want to deal with a distracting dim in the background.

Note that when you leave a special workspace, the workspace in the background will dim inactive windows if there\*(Aqs more than one window. If you want to avoid this, use this option in combination with ignore_leaving_special.

Note that if you\*(Aqre watching a video you never want to dim, you can use windowrules in Hyprland to achieve this.
.TP
\fB\-I\fR, \fB\-\-ignore\-leaving\-special\fR
Don\*(Aqt dim when closing a special workspace

This is useful if you have multiple windows in your main workspace and usually don\*(Aqt switch between them.
.TP
\fB\-D\fR, \fB\-\-dialog\-dim\fR=\fISTRENGTH\fR
Dim windows if they\*(Aqre the same class and floating (strength_default: 0.7)

This option is particularly useful for dimming dialog boxes started by applications since those tend to have the same class and be floating.

The dim is a permanent dim while working in the floating window of the same class.

Note that the dim is removed when switching workspaces or doing any other event.

Optionally specify a strength value to change how much dim is applied to dialog windows. The default strength value is 0.7.
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Show information about what hyprdim is doing
.TP
\fB\-h\fR, \fB\-\-help\fR
Print help (see a summary with \*(Aq\-h\*(Aq)
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version
.SH VERSION
v2.2.1
.SH AUTHORS
Copyright (C) 2023 Donovan Glover <https://donovan.is/>


completions/hyprdim.bash
_hyprdim() {
    local i cur prev opts cmd
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="hyprdim"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        hyprdim)
            opts="-s -d -f -b -p -n -i -I -D -v -h -V --strength --duration --fade --bezier --persist --no-dim-when-only --ignore-entering-special --ignore-leaving-special --dialog-dim --verbose --help --version"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --strength)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --duration)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -d)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --fade)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -f)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --bezier)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -b)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --dialog-dim)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -D)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

complete -F _hyprdim -o nosort -o bashdefault -o default hyprdim
